File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        int n = nums.length;
        List<List<Integer>> ans = new ArrayList<>();
        
        for (int i = 0; i < (1 << n); i++) {
            List<Integer> temp = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if ( (i & (1 << j)) != 0 ) {
                    temp.add(nums[j]);
                }
            }
            ans.add(temp);
        }

        return ans;
    }
}

File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        int n = nums.size();

        for(int i = 0; i < (1 << n); i++)
        {
            vector<int> st;
            for(int j = 0; j < n; j++)
            {
                if( (i & (1 << j)) )
                    st.push_back(nums[j]);
            }
            ans.push_back(st);
        }
        sort(ans.begin(), ans.end());
        return ans;
    }
};